straxen/corrections_services.py

```diff
@@ -0,0 +1,222 @@
+        if xenon1t:
+            df_global = self.interface.read('global_xenon1t')
+        else:
+            df_global = self.interface.read('global')
+        try:
+            for correction, version in df_global.iloc[-1][global_version].items():
+                if 'elife' in correction:
+                    df = self.interface.read(correction)
+                    df = self.interface.interpolate(df, when)
+        except KeyError:
+            raise ValueError(f'Global version {global_version} not found')
+
+        return df.loc[df.index == when, global_version].values[0]
+
+    def get_pmt_gains(self, run_id, global_version='v1', xenon1t=False):
+        """
+        Smart logic to return pmt gains to PE values.
+        :param run_id: run id from runDB
+        :param global_version: global version
+        :param xenon1t: boolean, whether you are processing xenon1t data or not
+        :retrun: array of pmt gains to PE values
+        """
+        when = CmtServices.get_time(run_id, xenon1t)
+        if xenon1t:
+            df_global = self.interface.read('global_xenon1t')
+        else:
+            df_global = self.interface.read('global')
+        try:
+            # equivalent to 'to_pe' in gains_model
+            gains = []
+            for correction, version in df_global.iloc[-1][global_version].items():
+                if 'pmt' in correction:
+                    df = self.interface.read(correction)
+                    df = self.interface.interpolate(df, when)
+                    gains.append(df.loc[df.index == when, version].values[0])
+                pmt_gains = np.asarray(gains, dtype=np.float32)
+
+        except KeyError:
+            raise ValueError(f'Global version {global_version} not found')
+        # be cautious with very early runs 
+        test = np.isnan(pmt_gains)
+        if test.all:
+            raise ValueError(f'to_pe(PMT gains) values are NaN, no data available'
+                             f' for {run_id} in the gain model with version {global_version},'
+                             f' please set a cte values for {run_id}')
+            
+        return pmt_gains
+
+    def get_lce(self, run_id, s, position, global_version='v1', xenon1t=False):
+        """
+        Smart logic to return light collection eff map values.
+        :param run_id: run id from runDB
+        :param s: S1 map or S2 map
+        :param postion: event position
+        :param xenon1t: boolean, whether you are processing xenon1t data or not
+        """
+        raise NotImplementedError
+
+    def get_fdc(slef, run_id, position, global_version='v1', xenon1t=False):
+        """
+        Smart logic to return field distortion map values.
+        :param run_id: run id from runDB
+        :param postion: event position
+        :param xenon1t: boolean xenon1t data=False/True
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def get_time(run_id, xenon1t=False):
```

Comment:
Maybe we should make this function stand alone and put it into common.py since also other projects could benefit from it, what do you think?

Reply:
can be a good idea, not sure how many would use it